### ISSUE TITLE
Implement Image Search tags

### ISSUE THREAD

**Issue by dakom** [ji-devs/ji-cloud#1459]
blocked on:

* https://github.com/ji-devs/ji-cloud/issues/1391
* https://github.com/ji-devs/ji-cloud/issues/1086

**Comment by dakom** [comment]
This is two separate features really:

1. Filter for tags
2. Prioritize by tags

Both of these are just a `Vec<TagId>` on [ImageSearchQuery](https://docs.sandbox.jicloud.org/crate/shared/domain/image/struct.ImageSearchQuery.html)

The component itself should accept each of these as-is (the exact lists will be at https://docs.google.com/document/d/14d-yE1sc52hdl7b0EQnN1a7hq0KYwYr1S-GXDCurGbA/edit?usp=sharing)

In terms of implementation, it should be stored in the component as a `HashSet` and then turned into a Vec for the request so that we can freely add to the list without worry of duplication.

There is a special case where the list is handled internally:

## UI Checkbox

The API should provide an `Option` for this checkbox which _internally_ translates into appending the above HashSets based on an enum:

```
enum {
  BackgroundLayer1Filter, // adds TagId::BackgroundLayer1 to the image_tags filter
  BackgroundLayer2Filter, // adds TagId::BackgroundLayer2 to the image_tags filter
  StickersFilter, //sets `kind` to Some(ImageKind::Sticker)
}
```

If `None` is supplied then the checkbox shouldn't be shown at all.

_Note: the text for `StickersFilter` might be a _bit_ wrong since it may say something like "do not include backgrounds", but _technically_ it's filtering based on the `kind` instead of `image_tag`... to implement it as a proper inverse of the tag filters (i.e. "not these filters") we'd need extra capability on the backend, so we're just using `kind` for now. This works since our _only_ `ImageKind::Canvas` images are, indeed, backgrounds, for now...

**Comment by dakom** [comment]
Strings:

```rust
BackgroundLayer1Filter | BackgroundLayer2Filter => "Show only backgrounds",
StickersFilter => "Do not include backgrounds"
```

**Comment by MendyBerger** [comment]
By default, are those checkboxes checked?

**Comment by dakom** [comment]
> By default, are those checkboxes checked?

I think that's a question for @corinnewo

**Comment by corinnewo** [comment]
Yes, the default is checked - so on background tab, default is show only backgrounds.

**Comment by MendyBerger** [comment]
Resolved in 4e60927ba1e66b331f9edeafdac304288a2b5781

**Comment by dakom** [comment]
@corinnewo heads up - it won't be ready for QA until I go through and add it to the modules

For the checkbox, issue is here: https://github.com/ji-devs/ji-cloud/issues/1508
